enh-ruby-mode.el

```diff
@@ -1439,6 +1436,42 @@ With ARG, do it that many times."
       (setq erm-full-parse-p t)
       (error "%s" (substring response 1))))))
 
+(defun erm--end-p ()
+  "Is point directly after a block closing \"end\""
+  (let ((end-pos (- (point) 3)))
+    (and (>= end-pos (point-min)) (string= "end" (buffer-substring end-pos (point))) (eq (get-text-property end-pos 'indent) 'e))))
```

Comment:
This needs tests. Should be easy to do. Use existing tests as a template.

Reply:
I have added unit tests for the parent function which cover this functionality as well.